libsolidity/ast/AST.cpp

```diff
@@ -338,8 +338,14 @@ TypePointer FunctionDefinition::type() const
 TypePointer FunctionDefinition::typeViaContractName() const
 {
 	if (annotation().contract->isLibrary())
-		return FunctionType(*this).asCallableFunction(true);
-	return TypeProvider::function(*this, FunctionType::Kind::Declaration);
+	{
+		if (isPublic())
+			return FunctionType(*this).asCallableFunction(true);
```

Comment:
I think it makes sense to rename `asCallableFunction` to `asExternallyCallableFunction`